zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/authorization/PersistedAuthorization.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.state.authorization;
+
+import static io.camunda.zeebe.util.buffer.BufferUtil.bufferAsString;
+
+import io.camunda.zeebe.db.DbValue;
+import io.camunda.zeebe.msgpack.UnpackedObject;
+import io.camunda.zeebe.msgpack.property.ArrayProperty;
+import io.camunda.zeebe.msgpack.property.EnumProperty;
+import io.camunda.zeebe.msgpack.property.LongProperty;
+import io.camunda.zeebe.msgpack.property.StringProperty;
+import io.camunda.zeebe.msgpack.value.StringValue;
+import io.camunda.zeebe.protocol.impl.record.value.authorization.AuthorizationRecord;
+import io.camunda.zeebe.protocol.record.value.AuthorizationOwnerType;
+import io.camunda.zeebe.protocol.record.value.AuthorizationResourceType;
+import io.camunda.zeebe.protocol.record.value.PermissionType;
+import io.camunda.zeebe.util.buffer.BufferUtil;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+public class PersistedAuthorization extends UnpackedObject implements DbValue {
+
+  private final LongProperty authorizationKeyProp = new LongProperty("authorizationKey");
+  private final StringProperty ownerIdProp = new StringProperty("ownerId", "");
+  private final EnumProperty<AuthorizationOwnerType> ownerTypeProp =
+      new EnumProperty<>(
+          "ownerType", AuthorizationOwnerType.class, AuthorizationOwnerType.UNSPECIFIED);
+  private final StringProperty resourceIdProp = new StringProperty("resourceId");
+  private final EnumProperty<AuthorizationResourceType> resourceTypeProp =
+      new EnumProperty<>("resourceType", AuthorizationResourceType.class);
+  private final ArrayProperty<StringValue> authorizationPermissionsProp =
```

Comment:
🔧 I'd rename this field `permissionsProp` and the property as well. In the AuthorizationRecord this field will be renamed as well